driver.go

```diff
@@ -57,48 +58,116 @@ func (s *Session) Logf(format string, a ...interface{}) {
 	fmt.Println(fmt.Sprintf(format+":%v", a))
 }
 
+// CreateGraph creates a graph and returns the neo4j ID of the root node
+// func CreateGraph(session *Session, tx neo4j.Transaction, nodes []graph.Node, config Config) (int64, error) {
+// 	nodeIds := make(map[graph.Node]int64)
+// 	allNodes := make(map[graph.Node]struct{})
+// 	//entityNodes := make(map[graph.Node]struct{})
+// 	// Find triples:
+// 	for _, node := range nodes {
+// 		allNodes[node] = struct{}{}
+// 	}
+// 	start := time.Now()
+// 	for node := range allNodes {
+
+// 		if _, exists := node.GetProperty(ls.EntitySchemaTerm); exists {
+// 			// TODO: Load entity nodes here
+// 			id := ls.AsPropertyValue(node.GetProperty(ls.EntityIDTerm)).AsString()
+// 			if len(id) > 0 {
+// 				if exists, nid, err := session.existsDB(tx, node, config); exists && err == nil {
+// 					nodeIds[node] = nid
+// 				}
+// 			}
+// 		}
+
+// 	}
+// 	duration := time.Since(start)
+// 	fmt.Println(fmt.Sprintf("time elapsed for existsDB: %v", duration))
+
+// 	start = time.Now()
+// 	for node := range allNodes {
+// 		hasEdges := false
+// 		for edges := node.GetEdges(graph.OutgoingEdge); edges.Next(); {
+// 			edge := edges.Edge()
+// 			if _, exists := allNodes[edge.GetTo()]; exists {
+// 				// Triple: edge.GetFrom(), edge, edge.GetTo()
+// 				if err := session.processTriple(tx, edge, nodeIds, config); err != nil {
+// 					return 0, err
+// 				}
+// 				hasEdges = true
+// 			}
+// 		}
+// 		if !hasEdges {
+// 			if _, exists := nodeIds[node]; !exists {
+// 				c := createNode{Config: config, node: node}
+// 				if err := c.Run(tx, nodeIds); err != nil {
+// 					return 0, err
+// 				}
+// 			}
+// 		}
+
+// 	}
+// 	duration = time.Since(start)
+// 	fmt.Println(fmt.Sprintf("time elapsed for processTriple: %v", duration))
+// 	return 0, nil
+// }
+
 // CreateGraph creates a graph and returns the neo4j ID of the root node
 func CreateGraph(session *Session, tx neo4j.Transaction, nodes []graph.Node, config Config) (int64, error) {
```

Comment:
I'm guessing you are planning to change CreateGraph into SaveGraph. Then, change the arg nodes to rootNodes, that is a more accurate name